### PR TITLE
fix: small text link adjustments

### PR DESCRIPTION
**Pull request by suevalov**
# Purpose of PR

Fixes small regressions that were found while integrating the TextLink component.

### CODE REVIEW

File: packages/components/text-link/src/TextLink.tsx [hunk 1]
```diff
@@ -109,7 +109,18 @@ const TextLink: PolymorphicComponentWithRef<
...
+      {...commonProps}
+      onClick={
+        isDisabled
+          ? (e) => {
+              e.preventDefault();
+            }
+          : commonProps.onClick
+      }
+      href={href}
+      {...(isDisabled ? { tabIndex: -1 } : {})}
```
Comment (suevalov) [inline packages/components/text-link/src/TextLink.tsx:122]: so, disabled anchor cannot be focused

File: packages/components/text-link/src/TextLink.tsx [hunk 2]
```diff
@@ -109,7 +109,18 @@ const TextLink: PolymorphicComponentWithRef<
   }
 
   return (
-    <a {...commonProps} href={href}>
+    <a
+      {...commonProps}
+      onClick={
```
Comment (suevalov) [inline packages/components/text-link/src/TextLink.tsx:114]: so disabled anchor cannot be clicked

Comment (denkristoffer) [reply packages/components/text-link/src/TextLink.tsx:114]: Didn't we render disabled TextLink as `span` before?

Comment (suevalov) [reply packages/components/text-link/src/TextLink.tsx:114]: no, we didn't

File: packages/components/icons/src/Gift.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { generateIcon } from '@contentful/f36-icon';
 
 export const Gift = generateIcon({
   name: 'Gift',
+  viewBox: '0 0 18 18',
```
Comment (burakukula) [inline packages/components/icons/src/Gift.tsx:6]: 🤔 why do we need this viewBox?

Comment (suevalov) [reply packages/components/icons/src/Gift.tsx:6]: because it was like that in v3 and it wasn't converted properly

**Review by burakukula — APPROVED** [review]
LGTM

File: packages/components/text-link/src/TextLink.tsx (hunk 2 above)
Comment (denkristoffer) [reply packages/components/text-link/src/TextLink.tsx:114]: Okay, well that would be my suggestion here to get these behaviours automatically.

**Review by gui-santos — APPROVED** [review]

File: packages/components/text-link/src/TextLink.tsx (hunk 2 above)
Comment (suevalov) [reply packages/components/text-link/src/TextLink.tsx:114]: I guess we'll have some typings problems in this case and we'd need to filter out some properties that are related only to `a` tag. I'd rather keep it this way.

Comment (gui-santos) [reply packages/components/text-link/src/TextLink.tsx:114]: Oh! I already approved the PR, but this ☝️  seems like a good idea

**Review by massao — APPROVED** [review]

File: packages/components/text-link/src/TextLink.tsx (hunk 2 above)
Comment (suevalov) [reply packages/components/text-link/src/TextLink.tsx:114]: one more argument against it: we would still need to remove `onClick` handler on `span` element.

**Review by bgutsol — APPROVED** [review]
All good!
My only comment is to notify @domarku to approve this design changes (icon size and margin) and update Figma files, so we keep design and components in consistency

**Comment by domarku** [comment]
Could we wait for the merge until later in the day? I want to understand the initial design decision from @fabe before we make the change. Thanks!

**Comment by suevalov** [comment]
@domarku yes, we can

File: packages/components/text-link/src/TextLink.tsx (hunk 2 above)
Comment (denkristoffer) [reply packages/components/text-link/src/TextLink.tsx:114]: I agree it's pretty bothersome for types 🙁

**Comment by domarku** [comment]
Change approved!

**Pull request merged**